ballerina/response_generator.bal

```diff
@@ -43,16 +43,16 @@ class ResponseGenerator {
         if parentValue is Scalar || parentValue is Scalar[] {
             return parentValue;
         }
-        if parentValue is map<anydata> {
-            if isMap(parentValue) {
-                return self.getResultFromMap(<map<anydata>>parentValue, parentNode);
-            }
+        if parentValue is map<anydata> && isMap(parentValue) {
+            return self.getResultFromMap(parentValue, parentNode);
+        }
+        if parentValue is map<any> {
             return self.getResultFromRecord(parentValue, parentNode);
         }
```

Comment:
This condition does not look right. 

The `is map<anydata>` returns true for both `map` and `record` types. That's why we need an additional check `isMap()` using native code. Now, in this case, we have 4 cases. 

- `map` with `anydata`
- `map` with `any`
- `record` with `anydata`
- `record` with `any`

But I think we can treat `anydata` and `any` similarly here. So, what we really need is 

```suggestion
        if parentValue is map<any> {
            if isMap(parentValue) {
                return self.getResultFromMap(parentValue, parentNode);
            }
            return self.getResultFromRecord(parentValue, parentNode);
        }
```

Reply:
Fixed with [93433b7](https://github.com/ballerina-platform/module-ballerina-graphql/pull/1024/commits/93433b74c3f72d4ab79e469bae9bf9a73c58114b)